packages/accounts-controller/src/AccountsController.ts

```diff
@@ -776,6 +789,20 @@ export class AccountsController extends BaseController<
     });
   }
 
+  /**
+   * Retrieves the value of a specific metadata key for an existing account.
+   * @param accountId - The ID of the account.
+   * @param metadataKey - The key of the metadata to retrieve.
+   * @returns The value of the specified metadata key, or undefined if the account or metadata key does not exist.
+   */
+  #populateExistingMetadata<T extends keyof InternalAccount['metadata']>(
+    accountId: string,
+    metadataKey: T,
+  ): InternalAccount['metadata'][T] | undefined {
+    const internalAccount = this.getAccount(accountId);
+    return internalAccount ? internalAccount.metadata[metadataKey] : undefined;
```

Comment:
Normally `getAccount` will return either an `InternalAccount` or `undefined` so I guess we can use this notation:

```suggestion
    return internalAccount?.metadata[metadataKey];
```

If you find it less clear, we can keep yours!